### PR TITLE
Adding changes for the search client side of the pipeline

### PR DESCRIPTION
**Pull request by dyshi**
Code containing changes to searchers and feature extractors

### CODE REVIEW

File: src/main/java/io/anserini/ltr/BaseFeatureExtractor.java [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+    }
+    /**
+     * Method used to print a line of feature vector to the output file
+     * @param out           The output stream
+     * @param qid           Qrel Id
+     * @param qrel          The Qrel relevance value
+     * @param docId         The stored Doc Id
+     * @param features      The feature vector in featureNum:value form
+     */
+    public static void writeFeatureVector(PrintStream out, String qid, int qrel, String docId, float[] features) {
```
Comment (lintool) [inline src/main/java/io/anserini/ltr/BaseFeatureExtractor.java]: consistent 2-space indent everywhere?

File: src/main/java/io/anserini/search/SearchWebCollection.java [hunk 2]
```diff
@@ -284,39 +267,57 @@ public static void main(String[] args) throws IOException, ParseException {
...
+    } else {
+      LOG.info("Using default FSDirectory");
+      dir = FSDirectory.open(Paths.get(searchArgs.index));
+    }
+
     Similarity similarity = null;
 
     if (searchArgs.ql) {
-      LOG.info("Using QL scoring model with mu=" + searchArgs.mu);
+      LOG.info("Using QL scoring model");
```
Comment (lintool) [inline src/main/java/io/anserini/search/SearchWebCollection.java]: restore printing out of the parameters

File: src/main/java/io/anserini/search/SearchWebCollection.java [hunk 3]
```diff
@@ -240,38 +252,9 @@ public void search(SortedMap<Integer, String> topics, String submissionFile, Sim
     out.close();
   }
 
-  public void search(RerankerCascade cascade, SortedMap<Integer, String> topics, String submissionFile, Similarity similarity, int numHits, final String runTag) throws IOException {
```
Comment (lintool) [inline src/main/java/io/anserini/search/SearchWebCollection.java]: I may be getting lost in the diffs, but where did this method go?

**Comment by lintool** [comment]
Mostly looks fine, just a few minor changes.

File: src/main/java/io/anserini/search/SearchWebCollection.java (hunk 3 above)
Comment (dyshi) [inline src/main/java/io/anserini/search/SearchWebCollection.java]: Oh the search method with that signature wasn't used anywhere, should I change it back?

Comment (dyshi) [inline src/main/java/io/anserini/search/SearchWebCollection.java]: I've added back the signature that doesn't require a reranker cascade.

**Pull request closed without merging**